commands/flags.go

```diff
@@ -57,23 +58,48 @@ func parseOptions(cliContext *cli.Context) (*options.Options, error) {
 	}
 	logger.SetLevel(level)
 
-	ports := cliContext.IntSlice("port")
-	if len(ports) == 0 {
-		return nil, MissingParam(portFlag.Name)
-	}
-
 	listener := cliContext.String("listener")
 	if listener == "" {
 		return nil, MissingParam(listenerFlag.Name)
 	}
 
+	checksFile := cliContext.String("checks")
+	if checksFile == "" {
+		return nil, MissingParam(checksFlag.Name)
+	}
+
+	checks, err := parseChecksFile(checksFile)
+	if err != nil {
+		return nil, err
+	}
+
 	return &options.Options{
-		Ports:          ports,
+		Checks:         checks,
 		Listener:       listener,
 		Logger:         logger,
 	}, nil
 }
 
+func parseChecksFile(checksFile string) (*options.Checks, error) {
+	checksFileContents, err := ioutil.ReadFile(checksFile)
+	if err != nil {
+		return nil, err
+	}
+
+	var checks options.Checks
+
+	err = yaml.Unmarshal(checksFileContents, &checks)
+	if err != nil{
+		return nil, err
+	}
+
+	if len(checks.TcpChecks) + len(checks.HttpChecks) + len(checks.ScriptChecks) == 0 {
+		return nil, err
```

Comment:
Don't return the previous error (which will be `nil`), create a new one

Reply:
good catch!